contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -203,6 +210,22 @@ class async_protocol_handler
       }
       return m_timer_cancelled;
     }
+    virtual void reset_timer()
+    {
+      boost::system::error_code ignored_ec;
+      m_timer.cancel(ignored_ec);
+      m_timer.expires_from_now(boost::posix_time::milliseconds(m_timeout));
+      m_timer.async_wait([this](const boost::system::error_code& ec)
+      {
+        if(ec == boost::asio::error::operation_aborted)
+          return;
+        MINFO(m_con.get_context_ref() << "Timeout on invoke operation happened, command: " << m_command << " timeout: " << m_timeout);
+        std::string fake;
+        m_cb(LEVIN_ERROR_CONNECTION_TIMEDOUT, fake, m_con.get_context_ref());
+        m_con.close();
+        m_con.finish_outer_call();
+      });
```

Comment:
I think this also ought to check m_cancel_timer_called to avoid resetting in that case. It's not clear whether that can be called from another thread, though the lack of locking hints it can't...